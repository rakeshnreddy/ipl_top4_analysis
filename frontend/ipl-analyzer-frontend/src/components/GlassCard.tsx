import React, { type ReactNode } from 'react';
import styles from './GlassCard.module.css';

interface GlassCardProps {
  children: ReactNode;
  className?: string;
  // Add other props like 'as' for polymorphic component if needed later
}

const GlassCard: React.FC<GlassCardProps> = ({ children, className = '' }) => {
  const combinedClassName = `${styles.card} ${className}`.trim();

  return (
    <div className={combinedClassName}>
      {children}
    </div>
  );
};

export default GlassCard;
