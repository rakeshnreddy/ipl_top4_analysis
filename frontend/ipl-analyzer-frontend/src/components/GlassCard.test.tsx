/// <reference types="vitest/globals" />
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import GlassCard from './GlassCard';
import styles from './GlassCard.module.css'; // To check against actual class name

describe('GlassCard Component', () => {
  it('Test 1: Renders children correctly', () => {
    render(
      <GlassCard>
        <p>Test Child</p>
      </GlassCard>
    );
    expect(screen.getByText('Test Child')).toBeInTheDocument();
  });

  it('Test 2: Applies the base card style', () => {
    const { container } = render(<GlassCard>Content</GlassCard>);
    // Check if the first child (the div rendered by GlassCard) has the specific module class name
    expect(container.firstChild).toHaveClass(styles.card);
  });

  it('Test 3: Applies additional className when provided', () => {
    const customClass = 'my-custom-card-style';
    const { container } = render(<GlassCard className={customClass}>Content</GlassCard>);
    expect(container.firstChild).toHaveClass(styles.card);
    expect(container.firstChild).toHaveClass(customClass);
  });

  it('Test 4: Renders correctly and matches snapshot', () => {
    const { container } = render(
      <GlassCard className="extra-class">
        <span>Snapshot Content</span>
      </GlassCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
