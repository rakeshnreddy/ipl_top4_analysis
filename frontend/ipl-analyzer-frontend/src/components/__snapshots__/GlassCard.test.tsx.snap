// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GlassCard Component > Test 4: Renders correctly and matches snapshot 1`] = `
<div
  class="_card_6217ab extra-class"
>
  <span>
    Snapshot Content
  </span>
</div>
`;
